gcc 11.2 warnings

Below a non-SIMD build.
gcc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)


```
In file included from truecrypt_fmt_plug.c:61:
truecrypt_fmt_plug.c: In function 'crypt_all._omp_fn.0':
pbkdf2_hmac_sha512.h:55:23: error: 'lens[0]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   55 |         for (i = 0; i < KL; i++) {
      |                     ~~^~~~
truecrypt_fmt_plug.c:481:21: note: 'lens[0]' was declared here
  481 |                 int lens[INNER_BATCH_MAX_SZ];
      |                     ^~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:1556: truecrypt_fmt_plug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:190: default] Error 2

Exited with code exit status 2
```

--- a/src/truecrypt_fmt_plug.c
+++ b/src/truecrypt_fmt_plug.c
@@ -460,15 +460,17 @@ static int decrypt_and_verify(unsigned char *key, int algorithm)
 
 static int crypt_all(int *pcount, struct db_salt *salt)
 {
-	int i, inner_batch_size = 1;
+	int i;
 	const int count = *pcount;
 
 #if SSE_GROUP_SZ_SHA512
 #define INNER_BATCH_MAX_SZ SSE_GROUP_SZ_SHA512
+	int inner_batch_size = 1;
 	if (psalt->hash_type == IS_SHA512)
 		inner_batch_size = SSE_GROUP_SZ_SHA512;
 #else
 #define INNER_BATCH_MAX_SZ 1
+#define inner_batch_size 1
 #endif
 
 	memset(cracked, 0, sizeof(cracked[0]) * count);

